collision.go

```diff
@@ -28,12 +29,46 @@ func (sc *SpaceComponent) Center(p Point) {
 }
 
 func (sc SpaceComponent) AABB() AABB {
```

Comment:
This one is indeed nice to document. Most of the rest will come whenever we decide to work on #117. 

Done.